pkg/kn/commands/source/apiserver/flags.go

```diff
@@ -41,27 +41,44 @@ type APIServerSourceUpdateFlags struct {
 }
 
 type resourceSpec struct {
-	kind         string
-	apiVersion   string
-	isController bool
+	Kind         string
+	ApiVersion   string
+	IsController bool
 }
 
-// GetAPIServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true,Pod:v2:false
-func (f *APIServerSourceUpdateFlags) GetAPIServerResourceArray() (*[]v1alpha1.ApiServerResource, error) {
-	var resourceList []v1alpha1.ApiServerResource
+// getAPIServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true,Pod:v2:false
+func (f *APIServerSourceUpdateFlags) getAPIServerResourceArray() ([]resourceSpec, error) {
+	var resourceList []resourceSpec
 	for _, r := range f.Resources {
 		resourceSpec, err := getValidResource(r)
 		if err != nil {
 			return nil, err
 		}
-		resourceRef := v1alpha1.ApiServerResource{
-			APIVersion: resourceSpec.apiVersion,
-			Kind:       resourceSpec.kind,
-			Controller: resourceSpec.isController,
+		resourceList = append(resourceList, *resourceSpec)
+	}
+	return resourceList, nil
+}
+
+// getAPIServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true,Pod:v2:false
+func (f *APIServerSourceUpdateFlags) getUpdateAPIServerResourceArray() ([]resourceSpec, []resourceSpec, error) {
+	var added []resourceSpec
+	var removed []resourceSpec
+	for _, r := range f.Resources {
+		if strings.HasSuffix(r, "-") {
+			resourceSpec, err := getValidResource(r[:len(r)-1])
+			if err != nil {
+				return nil, nil, err
+			}
+			removed = append(removed, *resourceSpec)
```

Comment:
Lets get https://github.com/knative/client/pull/592 in first and update this PR accordingly.

Reply:
@daisy-ycguo : #592 is merged, please update the PR to use the shared utility to parse minus suffix.

Reply:
Thank you @navidshaikh 
resources at `apiserver update` are not key=value format. So it's not convenient to use ParseMinusSuffix here. But I added a common utility in parsing_helper.go called `AddListAndRemovalListFromArray` to parse minus suffix in an array.